src/registry_schemas/schemas/party.json

```diff
@@ -119,7 +120,48 @@
                 "officer",
                 "roles",
                 "mailingAddress"
-            ]
+            ],
+            "if": {
+                "allOf":[
+                    {
+                        "properties": {
+                            "roles": {
+                                "not":{
+                                    "contains":{
+                                        "type": "object",
+                                        "properties": {
+                                            "roleType": { "enum": ["Completing Party"] }
+                                        },
+                                        "required": ["roleType"]
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    {
+                        "properties":{
+                            "officer": {
+                                "type": "object",
+                                "properties": {
+                                    "partyType": { "enum": ["person"] }
+                                },
+                                "required":["partyType"]
+                            }
+                        }
+                    }
+                ]
+            },
+            "then": {
+                "properties": {
+                  "officer": {
+                    "type": "object",
+                    "properties": {
+                      "firstName": { "minLength": 1 }
+                    },
+                    "required": ["firstName"]
```

Comment:
i thought `lastName` should be the required field when the role is "Completing Party"